app/src/pages/inside/common/launchFiltersContainer/launchFiltersContainer.jsx

```diff
@@ -81,28 +86,37 @@ export class LaunchFiltersContainer extends Component {
       }, {});
 
   handleFilterChange = (conditions) => {
-    this.fetchLaunches(this.createFilterQuery(conditions));
-    this.updateFilter(
-      this.props.activeFilterId,
-      Object.keys(conditions).map((key) => ({ ...conditions[key] })),
-    );
+    const newFilter = this.createFilterQuery(conditions);
+    const currentFilter = this.createFilterQuery(this.getConditions());
+
+    if (!isEqual(currentFilter, newFilter)) {
+      this.fetchLaunches(newFilter);
+    }
+
+    if (this.props.activeFilter) {
+      this.updateFilter(
+        this.props.activeFilterId,
+        Object.keys(conditions).map((key) => ({ ...conditions[key] })),
+      );
+    } else {
+      this.props.createFilter();
```

Comment:
Please pass conditions here to avoid inconsistent state (these conditions are not reflected in redux state, only someshere in component's state).